apps/server/src/modules/room/api/test/room-invitation-link-update.api.spec.ts

```diff
@@ -0,0 +1,210 @@
+				const { loggedInClient } = await setup();
+				const someId = new ObjectId().toHexString();
+				const params: UpdateRoomInvitationLinkBodyParams = {
+					title: 'Room #101',
+					activeUntil: new Date(Date.now() + 1000000),
+					requiresConfirmation: true,
+					isOnlyForTeachers: true,
+					restrictedToCreatorSchool: true,
+				};
+				const response = await loggedInClient.put(someId, params);
+				expect(response.status).toBe(HttpStatus.FORBIDDEN);
+			});
+		});
+
+		describe('when id is not a valid mongo id', () => {
+			const setup = async () => {
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return a 400 error', async () => {
+				const { loggedInClient } = await setup();
+				const params = { title: 'Room #101', color: 'green' };
+				const response = await loggedInClient.put('42', params);
+				expect(response.status).toBe(HttpStatus.BAD_REQUEST);
+			});
+		});
+
+		describe('when the user has the required permissions', () => {
+			const setup = async () => {
+				const school = schoolEntityFactory.buildWithId();
+				const room = roomEntityFactory.build({ schoolId: school.id });
+				const roomInvitationLink = roomInvitationLinkEntityFactory.build({
+					roomId: room.id,
+					creatorSchoolId: school.id,
+				});
+				const { roomAdminRole } = RoomRolesTestFactory.createRoomRoles();
+				const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher({ school });
+				const userGroup = groupEntityFactory.buildWithId({
+					users: [{ role: roomAdminRole, user: teacherUser }],
+				});
+				const roomMembership = roomMembershipEntityFactory.build({
+					roomId: room.id,
+					userGroupId: userGroup.id,
+					schoolId: school.id,
+				});
+				await em.persistAndFlush([
+					room,
+					roomInvitationLink,
+					roomMembership,
+					teacherAccount,
+					teacherUser,
+					userGroup,
+					roomAdminRole,
+				]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(teacherAccount);
+
+				return { loggedInClient, room, roomInvitationLink };
+			};
+
+			describe('when the room invitation link does not exist', () => {
+				it('should return a 404 error', async () => {
+					const { loggedInClient } = await setup();
+					const someId = new ObjectId().toHexString();
+					const params: UpdateRoomInvitationLinkBodyParams = {
+						title: 'Room Inivitation renamed',
+						activeUntil: new Date(Date.now() + 1000000),
+						requiresConfirmation: true,
+						isOnlyForTeachers: true,
+						restrictedToCreatorSchool: true,
+					};
+
+					const response = await loggedInClient.put(someId, params);
+
+					expect(response.status).toBe(HttpStatus.NOT_FOUND);
+				});
+			});
+
+			describe('when the required parameters are given', () => {
```

Comment:
the given parameters and naming within this test scenario confuses me, but I guess they are still in progress ^^